api/src/oscc.c

```diff
@@ -1,55 +1,136 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <linux/can.h>
+#include <linux/can/raw.h>
 #include <net/if.h>
 #include <signal.h>
 #include <stdio.h>
+#include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/types.h>
+#include <sys/syscall.h>
 #include <unistd.h>
+#include <poll.h>
 
 #include "oscc.h"
 #include "internal/oscc.h"
 
 
-static int can_socket = -1;
+static int oscc_can_socket = UNINITIALIZED_SOCKET;
```

Comment:
This is probably an issue for you and @adrienthebo to decide but my preference is that globals don't like anything else, defines, local variables, parameters, etc.. Ideally they'd be loud, something like `GLOBAL_OSCC_CAN_SOCKET`.

Reply:
I was just sticking with the standard that was already here since I replaced can_socket. @adrienthebo Thoughts, opinions, etc?